core/math/a_star_grid_2d.h

```diff
@@ -111,18 +110,20 @@ class AStarGrid2D : public RefCounted {
 		}
 	};
 
+	LocalVector<bool> solid_mask;
 	LocalVector<LocalVector<Point>> points;
 	Point *end = nullptr;
 	Point *last_closest_point = nullptr;
 
 	uint64_t pass = 1;
 
 private: // Internal routines.
+	_FORCE_INLINE_ int32_t _to_mask_index(int32_t p_x, int32_t p_y) const {
```

Comment:
```suggestion
	_FORCE_INLINE_ size_t _to_mask_index(int32_t p_x, int32_t p_y) const {
```

Technically, arrays in C++ using `size_t` type to achieve index.